openpype/plugins/publish/extract_slate_global.py

```diff
@@ -0,0 +1,538 @@
+            if "timecode" in lower_line:
+                timecode = lower_line.split("timecode:")[1].strip()
+                self.log.debug("%s: Found timecode on iinfo output: %s", name, timecode)
+                break
+
+        timecode_frames = timecode_to_frames(timecode, self.data["fps"])
+        # Subtracts one frame to account for the slate frame
+        timecode_frames -= 1
+        timecode = frames_to_timecode(timecode_frames, self.data["fps"])
+        self.data["timecode"] = timecode
+        self.log.debug("%s: Timecode for slate set to: %s", name, timecode)
+
+        return timecode
+
+    def get_resolution_ffprobe(self, input):
+        """Find input resolution using ffprobe."""
+        try:
+            streams = get_ffprobe_streams(input, self.log)
+        except Exception as exc:
+            raise AssertionError(
+                "FFprobe couldn't read information about input file: '{}'.\n{}".format(
+                    input, str(exc)
+                )
+            )
+
+        # Try to find first stream with defined 'width' and 'height'
+        width = None
+        height = None
+        for stream in streams:
+            if "width" in stream and "height" in stream:
+                width = int(stream["width"])
+                height = int(stream["height"])
+                break
+
+        # Raise exception of any stream didn't define input resolution
+        if width is None:
+            raise AssertionError(
+                "FFprobe couldn't read resolution from input file: '{}'.".format(input)
+            )
+
+        return (width, height)
+
+
+class ExtractSlateGlobal(publish.Extractor):
+    """Extractor that creates slate frames from an HTML template using OIIO.
+
+    Slate frames are based on an html template that gets rendered
+    using headless chrome/chromium. They then get converted using oiio,
+    with an ffmpeg fallback (still not implemented).
+
+    It will work only on represenations having `slateGlobal = True` and
+    `tags` including `slate`
+    """
+
+    label = "Extract Slate Global"
+    order = pyblish.api.ExtractorOrder + 0.0305
+    families = ["slate"]
+    hosts = [
+        "nuke",
+        "maya",
+        "blender",
+        "houdini",
+        "shell",
+        "hiero",
+        "premiere",
+        "harmony",
+        "traypublisher",
+        "standalonepublisher",
+        "fusion",
+        "tvpaint",
+        "resolve",
+        "webpublisher",
+        "aftereffects",
+        "flame",
+        "unreal"
+    ]
+
+    _slate_data_name = "slateGlobal"
+
+    def process(self, instance):
+
+        if self._slate_data_name not in instance.data:
+            self.log.warning(
+                "Slate Global workflow is not active, skipping slate extraction..."
+            )
+            return
+
+        if "representations" not in instance.data:
+            self.log.error("No representations to work on!")
+            raise ValueError("No items in list.")
```

Comment:
Should this instead just log a warning and return early?

If not, let's at least report more clearly in the error what list is being referred to, etc.

Reply:
Yeah good call, changed to warning